### PR TITLE
Update to latest Django and Smartmin

### PR DESCRIPTION
**Pull request by rowanseymour**
Also no more guardian

### CODE REVIEW

File: temba/contacts/tests.py [hunk 1]
```diff
@@ -3884,7 +3884,7 @@ def request_export(query=''):
             return workbook.worksheets[0]
 
         # no group specified, so will default to 'All Contacts'
-        with self.assertNumQueries(39):
+        with self.assertNumQueries(38):
```
Comment (rowanseymour) [inline temba/contacts/tests.py]: Switching `ContactCRUDL.Export` from a `SmartXlsView` to a `SmartTemplateView` saves one potentially expensive query. SmartXlsView is designed to generate an excel spreadsheet on the fly so it fetches the list the contacts, a does a count. `ContactCRUDL.Export` however just triggers an export in a celery task so it doesn't need any object_list.

File: temba/orgs/migrations/0035_ensure_anon_user_exists.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.2 on 2017-06-14 10:41
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.db import migrations
+
+
+def ensure_anon_user_exists(apps, schema_editor):
```
Comment (rowanseymour) [inline temba/orgs/migrations/0035_ensure_anon_user_exists.py]: Existing dbs will have this user as created by django-guardian. Need to create it for clean and test dbs.

File: templates/frame.haml [hunk 3]
```diff
@@ -87,15 +87,11 @@
...
       %script{src:"{{ STATIC_URL }}bower/sticky-kit/jquery.sticky-kit.js"}
 
-
       // Non-bower packages
       %script{src:"{{ STATIC_URL }}lib/uuid.js"}
       %script{src:"{{ STATIC_URL }}lib/bootstrap-limit.js"}
       %script{src:"{{ STATIC_URL }}lib/jquery-ui-timepicker-addon.js"}
 
-      // smartmin
-      %script{src:"{{ STATIC_URL }}js/heavy_data.js"}
```
Comment (rowanseymour) [inline templates/frame.haml]: Best I can tell this isn't used anymore and is a leftover from django_select2

File: temba/utils/__init__.py [hunk 4]
```diff
@@ -494,3 +494,11 @@ def decode_base64(original):
...
 
     return decoded
+
+
+def get_anonymous_user():
+    """
+    Returns the anonymous user, originally created by django-guardian
+    """
+    from django.contrib.auth.models import User
+    return User.objects.get(username=settings.ANONYMOUS_USER_NAME)
```
Comment (norkans7) [inline temba/utils/__init__.py]: This will blow up for new installs without guardian, should we create one then?

Comment (rowanseymour) [inline temba/utils/__init__.py]: I added a migration for that

Comment (norkans7) [inline temba/utils/__init__.py]: Ah right

File: temba/contacts/tests.py (hunk 1 above)
Comment (norkans7) [inline temba/contacts/tests.py]: +1

**Comment by rowanseymour** [comment]
@norkans7 should definitely do that eventually tho I actually tried that not so long ago and ran into weirdness.

**Pull request merged**